lib/oauth/error-responder.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var _ = require('lodash');
+var url = require('url');
+
+var helpers = require('../helpers');
+var forms = require('../forms');
+var resolveStateToken = require('./common').resolveStateToken;
+
+/**
+ * oauthErrorResponder takes an error object and respondes
+ * either by rendering the login form with the error, or
+ * by returning the error as JSON.
+ *
+ * @method
+ *
+ * @param {Object} req - The http request.
+ * @param {Object} res - The http response.
+ * @param {Object} err - The error to handle.
+ */
+function oauthErrorResponder(req, res, err) {
+  var accepts = req.accepts(['html', 'json']);
+  var config = req.app.get('stormpathConfig');
+
+  function renderLoginFormWithError(err) {
+    var view = req.app.get('stormpathConfig').web.login.view;
+    var nextUri = url.parse(req.query.next || '').path;
+    var formActionUri = config.web.login.uri + (nextUri ? ('?next=' + nextUri) : '');
```

Comment:
Since this line is just copied from the login controller, I think that we should keep the same behaviour and try to refactor them at once (even better is to use the same code, and not copy). But this is too much work for the scope of this PR.

However, I did add the `encodeURIComponent`.